dbc2val/dbcfeederlib/canplayer.py

```diff
@@ -40,73 +38,42 @@ class CANplayer:
     """
 
     def __init__(self, dumpfile):
-        self.run = True
-        self.index = 0
-        self.indexout = 1
+        self.run = False
         self.messages = [can.message]
-        self.timestamp = None
+        self.dumpfile = dumpfile
 
+    def process_log(self):
         # open the file for reading can messages
-        log.info("Replaying candump from {}".format(dumpfile))
-        log_reader = can.LogReader(dumpfile)
-        # get all messages out of the dumpfile and store into array of can messages
+        log.info("Replaying CAN message log {}".format(self.dumpfile))
+        # using MessageSync in order to consider timestamps of CAN messages
+        # and the delays between them
+        log_reader = can.MessageSync(messages=can.LogReader(self.dumpfile), timestamps=True)
         for msg in log_reader:
-            # store the sum of messages
-            self.index = self.index + 1
-            # add message to array
-            self.messages.append(msg)
-            # log.debug("Message Read out of candump file: \n{}".format(msgFromLog))
-        log.debug("Parsing of can messages form the dump file is finished")
-
-    def start_replaying(self, canport):
-        log.debug("Using virtual bus to replay CAN messages (channel: %s)", canport)
-        self.bus = can.interface.Bus(bustype="virtual", channel=canport, bitrate=500000) # pylint: disable=abstract-class-instantiated
-        txThread = threading.Thread(target=self.txWorker)
-        txThread.start()
+            if not self.run:
+                break
+            try:
+                self.bus.send(msg)
+                log.debug(f"Message sent on {self.bus.channel_info}")
+                log.debug(f"Message: {msg}")
+            except can.CanError:
+                log.debug("Failed to send message via CAN bus")
 
-    def getNextMessage(self):
-        val = self.messages[self.indexout]
-        self.indexout += 1
-        # in case of end of array set the index pointer to start again
-        if self.indexout >= self.index:
-            self.indexout = 1
-        return val
+        log.debug("Replayed all messages from CAN log file")
```

Comment:
I think this could be a log.info, can be quite useful and shall only be given once, right?

Reply:
this is logged each time the log has been re-played fully.

Reply:
Ah, then it can actually come quite often if we run the log repeatedely